### PR TITLE
fix: add getMany and getManyWith

### PR DESCRIPTION
**Pull request by dignifiedquire**
Needs https://github.com/ipfs/js-ipfs-block-service/pull/21

### CODE REVIEW

File: src/dag-service.js [hunk 1]
```diff
@@ -25,33 +25,78 @@ module.exports = class DAGService {
 
   // get retrieves a DAGNode, using the Block Service
   get (multihash, callback) {
-    const isMhash = isIPFS.multihash(multihash)
-    const isPath = isIPFS.path(multihash)
+    this.getMany([multihash], (err, results) => {
```
Comment (hackergrrl) [inline src/dag-service.js:28]: Could you verify this was called with a multihash?

File: test/dag-service-test.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ module.exports = function (repo) {
       var encodedMh = 'QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG'
       dagService.get(encodedMh, (err, fetchedNode) => {
         expect(err).to.not.exist
+        console.log(fetchedNode)
```
Comment (hackergrrl) [inline test/dag-service-test.js:28]: debug output?

File: test/dag-service-test.js [hunk 3]
```diff
@@ -56,26 +57,23 @@ module.exports = function (repo) {
...
         done()
       })
     })
 
     it('supply something weird', (done) => {
       const mh = 3
       dagService.get(mh, (err, fetchedNode) => {
-        const error = 'Error: Invalid Key'
-        expect(err.toString()).to.equal(error)
+        expect(err.message).to.be.eql('Invalid Key')
```
Comment (hackergrrl) [inline test/dag-service-test.js:76]: Could you add tests for `getMany` and `getWithMany`?

File: src/dag-service.js [hunk 4]
```diff
@@ -25,33 +25,78 @@ module.exports = class DAGService {
...
+
+      if (results[key].error) {
+        return callback(results[key].error)
+      }
+
+      callback(null, results[key].block)
+    })
+  }
+
+  getWithMany (keys, callback) {
```
Comment (hackergrrl) [inline src/dag-service.js:79]: Could you document what this method does/accepts/outputs? Both of
- [ ] here (doc comment)
- [ ] in the readme (api listing)

same with `getMany`

File: src/dag-service.js [hunk 5]
```diff
@@ -25,33 +25,78 @@ module.exports = class DAGService {
...
-    this.bs.getBlock(formatted, (err, block) => {
+    this.getWithMany([key], (err, results) => {
       if (err) {
         return callback(err)
       }
 
-      const node = new DAGNode()
-      node.unMarshal(block.data)
-      return callback(null, node)
+      const key = Object.keys(results)[0]
```
Comment (hackergrrl) [inline src/dag-service.js:69]: What if `results` has no keys? (This'd make a good test case!)

(same with Line 33)

File: package.json [hunk 6]
```diff
@@ -57,4 +57,4 @@
     "rimraf": "^2.5.0",
     "run-series": "^1.1.4"
   }
-}
\ No newline at end of file
+}
```
Comment (hackergrrl) [inline package.json:60]: extraneous newline

File: src/dag-service.js [hunk 7]
```diff
@@ -25,33 +25,78 @@ module.exports = class DAGService {
...
-    if (isPath) {
-      const ipfsKey = multihash.replace('/ipfs/', '')
-      this.getWith(ipfsKey, callback)
-    }
+      callback(null, results[key])
+    })
+  }
+
+  getMany (mhs, callback) {
+    let err = false
```
Comment (hackergrrl) [inline src/dag-service.js:44]: Could you check that `mhs` is indeed a) an array, and b) contains multihashes?

File: package.json (hunk 6 above)
Comment (dignifiedquire) [inline package.json:60]: all files shall have newlines at the end :P

**Comment by dignifiedquire** [comment]
Closing in favor of #44

**Pull request closed without merging**